podpac/core/coordinates/affine_coordinates.py

```diff
@@ -0,0 +1,376 @@
+            return super(AffineCoordinates, self)._getsubset(index).simplify()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Properties
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def _coords(self):
+        if not hasattr(self, "_coords_"):
+            self._coords_ = [
+                ArrayCoordinates1d(c, name=dim) for c, dim in zip(self.coordinates.transpose(2, 0, 1), self.dims)
+            ]
+        return self._coords_
+
+    @property
+    def ndim(self):
+        return 2
+
+    @property
+    def affine(self):
+        """:rasterio.Affine: affine transformation for computing the coordinates from indexing values."""
+        return self._affine
+
+    @property
+    def dims(self):
+        return ("lat", "lon")
+
+    @property
+    def is_affine(self):
+        return True
+
+    @property
+    def origin(self):
+        origin = self.affine * [0, 0]
+        if self.dims == ("lat", "lon"):
+            origin = origin[::-1]
+        return origin
+
+    @property
+    def rotation(self):
+        return self.affine.rotation_angle
+
+    @property
+    def coordinates(self):
+        """ :tuple: computed coordinate values for each dimension. """
+
+        I = np.arange(self.shape[1]) + 0.5
+        J = np.arange(self.shape[0]) + 0.5
+        x, y = self.affine * np.meshgrid(I, J)
+        if self.dims == ("lat", "lon"):
+            c = np.stack([y, x])
+        else:
+            c = np.stack([x, y])
+        return c.transpose(1, 2, 0)
+
+    @property
+    def definition(self):
+        d = OrderedDict()
+        d["geotransform"] = self.geotransform
+        d["shape"] = self.shape
+        return d
+
+    @property
+    def full_definition(self):
+        return self.definition
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def copy(self):
+        """
+        Make a copy of the affine coordinates.
+
+        Returns
+        -------
+        :class:`AffineCoordinates`
+            Copy of the affine coordinates.
+        """
+        return AffineCoordinates(self.geotransform, self.shape)
+
+    def get_area_bounds(self, boundary):
+        """Get coordinate area bounds, including boundary information, for each unstacked dimension.
+
+        Arguments
+        ---------
+        boundary : dict
+            dictionary of boundary offsets for each unstacked dimension. Point dimensions can be omitted.
+
+        Returns
+        -------
+        area_bounds : dict
+            Dictionary of (low, high) coordinates area_bounds in each unstacked dimension
+        """
+
+        # TODO the boundary offsets need to be transformed
```

Comment:
These are always centered, I can implement this now.

Reply:
This isn't that simple, still working on it.

`Coordinates.get_area_bounds` loops thrugh the udims, so won't hit this. So that needs to be updated when the coordinates have affine coordinates.

Note that nothing uses this (see #422), so I think an exception when trying to get affine area bounds is okay for now. When we implement #422, we'll hit the exception, but until then, we don't need it.